api/s2n.h

```diff
@@ -1584,6 +1584,26 @@ S2N_API extern int s2n_client_hello_get_session_id(struct s2n_client_hello *ch,
 S2N_API extern int s2n_client_hello_get_supported_groups(struct s2n_client_hello *ch, uint16_t *groups,
         uint16_t groups_count_max, uint16_t *groups_count);
 
+/**
+ * Gets the length of the first server name in a Client Hello.
+ *
+ * @param ch A pointer to the ClientHello
+ * @param length A pointer which will be populated with the length of the server name
+ */
+S2N_API extern int s2n_client_hello_get_server_name_length(struct s2n_client_hello *ch, uint16_t *length);
```

Comment:
Could we make these API's use `const struct s2n_client_hello`? That would make the rust bindings a little bit safer

Reply:
I did look into this, unfortunately adding const to this parameter has cascading effects. Basically you have to then make the extension parsing structs const, and then those structs eventually touch stuffer functions, at which point you are kind of screwed because we can't refactor stuffers to be const, they need to be able to be written to. 🤷‍♀️ I dunno, I think our codebase is too dependent on stuffers to have const pointers. Although I don't think it's a bad exercise to explore this more.

Reply:
Are those stuffers part of the client_hello? If this method isn't actually const and is modifying things on the client hello, then we should probably make the Rust bindings take a `&mut`.

Reply:
This method definitely isn't modifying the client hello. However, I do use stuffers to read the data in the client hello. And in general, our stuffer functions can't be initialized with const data, because stuffers technically can be both read and written to.

I think what you would want is a new construct, like a const_stuffer, where it's only valid to call our read functions on it but not our write functions. But I think it would require duplicating all of our s2n_stuffer_read functions.